src/components/LegacyMediaAnalytics/createMediaAnalyticsTracker.js

```diff
@@ -0,0 +1,277 @@
+          });
+        }
+      });
+    } else {
+      Object.keys(context).forEach(key => {
+        customMetadata.push({
+          name: key,
+          value: context[key]
+        });
+      });
+    }
+    mediaDetails.customMetadata = customMetadata;
+  }
+
+  return mediaDetails;
+};
+
+const createHeartbeat = ({ mainPingInterval, playerState, trackEvent }) => {
+  // eslint-disable-next-line consistent-return
+  return () => {
+    const currentTime = Date.now();
+    if (
+      Math.abs(currentTime - playerState.latestTriggeredEvent) / 1000 >
+      mainPingInterval
+    ) {
+      return trackEvent({ eventType: "ping" });
+    }
+  };
+};
+
+const createGetInstance = ({ config, logger, mediaEventManager }) => {
+  const { mainPingInterval } = config.mediaCollection;
+
+  const trackerState = {
+    qoe: null,
+    lastPlayhead: 0,
+    latestTriggeredEvent: null
+  };
+  let ticker;
+  let deferSession = defer();
+
+  const updateLastTimeEventTriggered = () => {
+    trackerState.latestTriggeredEvent = Date.now();
+  };
+  const getEventType = ({ eventType }) => {
+    if (eventType === Event.BufferComplete) {
+      return `media.${Event.Play}`;
+    }
+    return `media.${eventType}`;
+  };
+
+  const trackEvent = ({ eventType, mediaDetails }) => {
+    return deferSession.promise
+      .then(sessionID => {
+        const xdm = {
+          eventType: getEventType({ eventType }),
+          mediaCollection: {
+            sessionID,
+            playhead: trackerState.lastPlayhead
+          }
+        };
+        deepAssign(xdm.mediaCollection, mediaDetails);
+        const event = mediaEventManager.createMediaEvent({ options: { xdm } });
+        return mediaEventManager.trackMediaEvent({ event }).then(() => {
+          updateLastTimeEventTriggered();
+        });
+      })
+      .catch(error => {
+        logger.info(error);
+      });
+  };
+
+  const heartbeatEngine = createHeartbeat({
+    mainPingInterval,
+    deferSession,
+    playerState: trackerState,
+    trackEvent
+  });
+
+  return {
+    trackSessionStart: (mediaObject, contextData = {}) => {
+      if (isEmptyObject(mediaObject)) {
+        deferSession.reject("invalid media object");
+        return {};
+      }
+
+      const mediaCollection = {
+        playhead: 0
+      };
+      const {
+        sessionDetails,
+        customMetadata
+      } = convertSessionDetailsMetadataToXDM(contextData);
+
+      deepAssign(
+        mediaCollection,
+        mediaObject,
+        { sessionDetails },
+        { customMetadata }
+      );
+      const event = mediaEventManager.createMediaSession({
+        xdm: {
+          mediaCollection
+        }
+      });
+
+      return mediaEventManager
+        .trackMediaSession({
+          event
+        })
+        .then(result => {
+          updateLastTimeEventTriggered();
+          deferSession.resolve(result.sessionId);
+
+          ticker = setInterval(() => {
+            heartbeatEngine();
+          }, 1000);
```

Comment:
I'll try this later. I like `setTimeout` over `setInterval`, in my opinion if there is a page with multiple videos on the page, it can fill-up the storage.